### ISSUE TITLE
MSE of test data

### ISSUE THREAD

**Issue by matmil1997** [aaxwaz/Multivariate-Time-Series-forecast-using-seq2seq-in-TensorFlow#7]
Hi,

I have been trying to replicate your code, and everything has generally been working fine, apart from one (nit-picky) thing. 

In the notebook on your repo, you obtain an MSE value of 0.336378205965, and the y values on your graph range from -1 to about 3.5. 

On your website [(here)](https://weiminwang.blog/2017/09/29/multivariate-time-series-forecast-using-seq2seq-in-tensorflow/#multivariate), however,  you get a graph ranging from 0 to over 400. 

Looking at your raw data range for pm2.5, the values seem to range from 0-1000. 

I was just wondering why does this apparently random scaling occur? Is there a way to prevent this scaling? 

I am asking because it is making my data predictions non-interpretable. 

Otherwise, thank you for your work on this notebook, it's very good and interesting!

**Comment by matmil1997** [comment]
I found a quick fix to this problem. 

I removed the below block of code and everything worked just fine:

`## z-score transform y
y_mean = y_train.mean()
y_std = y_train.std()
y_train = (y_train - y_mean) / y_std
y_test = (y_test - y_mean) / y_std`

By doing this, my MSE metric was improved, and by visual inspection, the model better fitted the actual data. In fact, it made my loss function converge much more quickly, at about 200 iterations, versus the 1000 I had been previously using, which even then weren't sufficient.

A suggestion I would make is rather than print the loss at time t in your loops, make a list of these losses and plot them. It will give a much better understanding of how your algorithm works. 

`losses = []
    
    #print("Training losses: ")
    for i in range(total_iterations):
        batch_input, batch_output = generate_train_samples(batch_size=batch_size)
        
        feed_dict = {rnn_model['enc_inp'][t]: batch_input[:,t] for t in range(input_seq_len)}
        feed_dict.update({rnn_model['target_seq'][t]: batch_output[:,t] for t in range(output_seq_len)})
        _, loss_t = sess.run([rnn_model['train_op'], rnn_model['loss']], feed_dict)
        #print(loss_t)
        losses.append(loss_t)
        
    plt.figure(figsize=(15,5))
    plt.plot(losses)
    plt.xlabel("Number of iterations")
    plt.ylabel("Loss")`

Closing the issue.
